### PR TITLE
RFS-95 - Stage 1 - Federation members with three keys

### PR DESCRIPTION
**Pull request by amendelzon**
Refactoring the `Federation` class so that it is now comprised of a list of `FederationMember` (new class) instances, each containing three secp256k1 public keys: one for BTC, one for RSK and one called MST with yet undefined usage, but intended as a master key (hence the name) for multiple usages in the future.

The rest of the components adapt to this new abstraction by instancing these new objects with the same key for each of the three required keys. Therefore, the behavior of the node doesn't really change at this stage.

It is important to mention that, for a federate node to work, one must add an "RSK" signer to the list of signers in the configuration, copying the already configured "BTC" signer, so that then both signers use the same key.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 12 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 7 minor

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [BridgeSerializationUtils.java#L28](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java#L28): Unused import - org.ethereum.crypto.ECKey. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L48](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L48): Unused import - org.ethereum.crypto.ECKey. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [BridgeSupport.java#L345](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L345): The Cyclomatic Complexity of this method "registerBtcTransaction" is 17 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BridgeSupport.java#L1288](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1288): Method co.rsk.peg.BridgeSupport.getFederatorPublicKey(int) calls String.format on a static (non parameterized) format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASPP_STATIC_FORMAT_STRING)
1. ![MAJOR][MAJOR] [BridgeSupport.java#L1288](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1288): Format-string method String.format(String, Object[]) called with format string "Federator index must be between 0 and {}" wants 0 arguments but is given 1 in co.rsk.peg.BridgeSupport.getFederatorPublicKey(int) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AVA_FORMAT_STRING_EXTRA_ARGUMENTS_PASSED)
1. ![MINOR][MINOR] [BridgeSerializationUtils.java#L258](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java#L258): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [Federation.java#L69](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/Federation.java#L69): Return a copy of "members". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2384)
1. ![MINOR][MINOR] [FederationMember.java#L53](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L53): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [FederationMember.java#L74](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L74): Move this variable to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [FederationMember.java#L91](https://github.com/rsksmart/rskj/blob/f5390a6f221a35fa8469909d79afc90ade9a82f9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L91): Move this constructor to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 8 issues

1. ![MAJOR][MAJOR] [BridgeSupport.java#L344](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L344): The Cyclomatic Complexity of this method "registerBtcTransaction" is 17 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MINOR][MINOR] [BridgeSerializationUtils.java#L257](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java#L257): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [Federation.java#L69](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/Federation.java#L69): Return a copy of "members". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2384)
1. ![MINOR][MINOR] [FederationMember.java#L54](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L54): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [FederationMember.java#L75](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L75): Move this variable to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [FederationMember.java#L92](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L92): Move this constructor to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [PendingFederation.java#L55](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/PendingFederation.java#L55): Return a copy of "members". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2384)
1. ![MINOR][MINOR] [PendingFederation.java#L60](https://github.com/rsksmart/rskj/blob/66cf8ac52bb95c17186c983afdd99a1f401805f4/rskj-core/src/main/java/co/rsk/peg/PendingFederation.java#L60): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 1]
```diff
@@ -1351,7 +1351,7 @@ public Integer getRetiringFederationThreshold() {
             return null;
         }
 
-        List<BtcECKey> publicKeys = retiringFederation.getPublicKeys();
+        List<BtcECKey> publicKeys = retiringFederation.getBtcPublicKeys();
 
         if (index < 0 || index >= publicKeys.size()) {
             throw new IndexOutOfBoundsException(String.format("Retiring federator index must be between 0 and {}", publicKeys.size() - 1));
```
Comment (pmprete) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: the {} should be %s in String.format. I think this kind of interpolation is in logs only in this proyect

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 2]
```diff
@@ -1681,7 +1686,7 @@ public Integer getPendingFederationSize() {
             return null;
         }
 
-        List<BtcECKey> publicKeys = currentPendingFederation.getPublicKeys();
+        List<BtcECKey> publicKeys = currentPendingFederation.getBtcPublicKeys();
 
         if (index < 0 || index >= publicKeys.size()) {
             throw new IndexOutOfBoundsException(String.format("Federator index must be between 0 and {}", publicKeys.size() - 1));
```
Comment (pmprete) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: same here

**Comment by rskops** [comment]
SonarQube analysis reported 8 issues

1. ![MAJOR][MAJOR] [BridgeSupport.java#L344](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L344): The Cyclomatic Complexity of this method "registerBtcTransaction" is 17 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MINOR][MINOR] [BridgeSerializationUtils.java#L257](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java#L257): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [Federation.java#L69](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/Federation.java#L69): Return a copy of "members". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2384)
1. ![MINOR][MINOR] [FederationMember.java#L54](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L54): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)
1. ![MINOR][MINOR] [FederationMember.java#L75](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L75): Move this variable to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [FederationMember.java#L92](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/FederationMember.java#L92): Move this constructor to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)
1. ![MINOR][MINOR] [PendingFederation.java#L55](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/PendingFederation.java#L55): Return a copy of "members". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2384)
1. ![MINOR][MINOR] [PendingFederation.java#L60](https://github.com/rsksmart/rskj/blob/e6413b27c8055177b6880db3d1f602bfa2525da9/rskj-core/src/main/java/co/rsk/peg/PendingFederation.java#L60): Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**